src/main/java/org/opensearch/neuralsearch/processor/combination/ScoreCombinationTechnique.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor.combination;
+
+import lombok.AllArgsConstructor;
+
+/**
+ * Collection of techniques for score combination
+ */
+@AllArgsConstructor
+public enum ScoreCombinationTechnique {
```

Comment:
String for technique passed by user at factory level, we can pass enum variable safer between that many layers. Plus it's a single place that encapsulates technique and method abstractions. I switched to this approach after trying technique and method or string and then method.

Reply:
Doesn't all you need is just calling combine method in the end?
Then, can we just pass `ArithmeticMeanScoreCombinationMethod` instance at factory level? I don't think passing instance is not safer than passing enum.

Reply:
my concern is more to keep factory light. We're going to add more methods and other configurations (like weights for sub-queries) later, and in such case factory became more responsible for kicking off calculations, while it's intended design was to parse user input and construct processor. Enum layer is taking that preparation for calculations responsibility so factory can stay light

Reply:
My point is that now it's a simple conversion String -> enum, and it's same as String -> method. But with planned next PR changes (and we know we'll be doing this as per [HLD RFC](https://github.com/opensearch-project/neural-search/issues/126)) it will be string + map_of_params -> method, and I'd like to keep it String -> enum. And also to keep knowledge of the actual computation method away from the factory and keep it at lower levels, factory knows just the technique, technique knows about method specifics .